### PR TITLE
fix res2net+dcn test bug

### PR DESCRIPTION
**Pull request by yuzhj**
I rewrite init_weights in the res2net.py to fix the test bug(AttributeError: 'Bottle2neck' object has no attribute 'conv2')

### CODE REVIEW

**Comment by yuzhj** [comment]
the format problem is so difficult!!!!!!!!!!

File: mmdet/models/backbones/res2net.py [hunk 1]
```diff
@@ -313,3 +317,34 @@ def make_res_layer(self, **kwargs):
...
+            for m in self.modules():
+                if isinstance(m, nn.Conv2d):
+                    kaiming_init(m)
+                elif isinstance(m, (_BatchNorm, nn.GroupNorm)):
+                    constant_init(m, 1)
+
+            if self.dcn is not None:
+                for m in self.modules():
+                    if isinstance(m, Bottle2neck):
+                        for n in m.convs:
```
Comment (xvjiarui) [inline mmdet/models/backbones/res2net.py:341]: We may add comments here.
Like `dcn in Res2Net bottleneck is in ModuleList`.

**Comment by xvjiarui** [comment]
> the format problem is so difficult!!!!!!!!!!

Hi @yuzhj 
Thanks for your contribution. You may check [contribute guide](https://github.com/open-mmlab/mmdetection/blob/master/.github/CONTRIBUTING.md) for reference.

File: mmdet/models/backbones/res2net.py (hunk 1 above)
Comment (yuzhj) [reply mmdet/models/backbones/res2net.py:341]: > We may add comments here.
> Like `dcn in Res2Net bottleneck is in ModuleList`.

ok, i will add comments

**Review by xvjiarui — APPROVED** [review]

**Pull request merged**